naturewatch_camera_server/ChangeDetector.py

```diff
@@ -142,48 +143,64 @@ def set_sensitivity(self, min_width, max_width):
 
     def stop_session(self):
-        self.logger.info('Ending photo capturing')
+        self.logger.info('ChangeDetector: ending capture')
+        if self.mode == "video":
+            self.camera_controller.stop_video_stream()
+        elif self.mode == "photo":
+            pass
         self.mode = "inactive"
+# TODO: required?
         self.session_start_time = self.get_fake_time()
 
+# TODO: whether to use the video-port or not does not directly depend on the mode
+# In case video is requested, the video port will always be used for both resolutions
+# In case photo is requested, the video port can be used, but need not. It should be left a matter of configuration
     def update(self):
         time.sleep(0.02)
-        if self.mode == "photo":
-            img = self.camera_controller.get_image()
-            if self.detect_change_contours(img) is True:
-                timestamp = self.get_formatted_time()
-                self.logger.info("ChangeDetector: Detected motion. Taking photo...")
-                self.file_saver.save_image(self.camera_controller.get_splitter_image(), timestamp)
-                self.file_saver.save_thumb(img, timestamp, self.mode)
-                self.lastPhotoTime = self.get_fake_time()
-        elif self.mode == "video":
-            self.camera_controller.wait_recording(1)
-            img = self.camera_controller.get_image()
-            if self.detect_change_contours(img) is True: 
-                self.logger.info("ChangeDetector: Detected motion. Capturing Video...")
-                timestamp = self.get_formatted_time()
-                self.file_saver.save_thumb(img, timestamp, self.mode)
-                try:
-                    start = self.get_fake_time()
-                    while self.get_fake_time() - start < self.config["video_duration_after_motion"]:
-                        self.camera_controller.wait_recording(1)
-                finally:
-                    self.logger.info("Video capture completed")
-                    with self.camera_controller.circularStream.lock:
-                        self.file_saver.save_video(self.camera_controller.circularStream, timestamp)
-                    self.lastPhotoTime = self.get_fake_time()
-                    self.logger.info("Video timer reset")
-
+        # only check for motion while a session is active
+        if self.mode in ["photo", "video"]:
+            # get an md image
+            img = self.camera_controller.get_md_image()
+            # only proceed if there is an image
+            if img is not None:
+                if self.detect_change_contours(img) is True:
+                    self.logger.info("ChangeDetector: detected motion. Starting capture...")
+                    timestamp = self.get_formatted_time()
+        # TODO: Thumb should be created from the photo itself, not the image that triggered the motion detection
+                    self.file_saver.save_thumb(img, timestamp, self.mode)
```

Comment:
This causes the thumbnail and the final photo to be different. We should pass the hi res image to the save_thumb function so that they look exactly the same. During a video capture, it doesn't really matter, so we could use the `img` var to create the thumbnail there. Sounds like `save_thumb` needs a bit of work to scale a full image down to the thumb size regardless of its original size. If you rather didn't do this for this PR, let me know and I'll accept this one, and I'll work on it instead.

Reply:
Yes, the difference isn't good. But that is basically as it was before my changes. This is the next thing I will work on. Will let you know if I get stuck need your help, OK?